libfreerdp/core/gateway/rdg.c

```diff
@@ -1335,7 +1336,7 @@ static BOOL rdg_process_handshake_response(rdpRdg* rdg, wStream* s)
 	         ", extendedAuth=%s",
 	         error, verMajor, verMinor, serverVersion, extended_auth_to_string(extendedAuth));
 
-	if (FAILED(errorCode))
+	if (FAILED((HRESULT)errorCode))
```

Comment:
ok, now I think we have seen all versions except the obvious (`HRESULT errorCode;`  instead of `UINT32 errorCode;` as definition :))
joking aside, sorry for the misleading last comments, the define in winpr is really faulty, seen the (very annoying) documentation in msdn. (don´t really like it as you can just throw in anything without the compiler complaining)

anyway, just leave it, well merge and do the rest from here on (to avoid further confusion, as mentioned before, need coffee ;))